app/controllers/tournaments_controller.rb

```diff
@@ -153,6 +154,41 @@ def create_event(event)
       xtables_id: xtables_id) unless Event.find_by(xtables_id: xtables_id)
   end
 
+  def generate_two_rounds_matchups(tournament)
+    # Check how many divisions there are
+    all_players = Player.where(tournament: tournament)
+    divisions = all_players.map { |player| player.division }.uniq
+
+    # Create players hash where each key is a division and
+    # each value is an array of the players in that division
+    players = {}
+    divisions.each do |div|
+      players[div] = Player.where(tournament: tournament).where(division: div).to_a
```

Comment:
Could this just be `Player.where(tournament: tournament, division: div).to_a` ?

Reply:
Answered elsewhere, but yes this should work but in my testing it didn't... for now what I have works